govcd/vm.go

```diff
@@ -456,3 +457,68 @@ func (vm *VM) Undeploy() (Task, error) {
 	return *task, nil
 
 }
+
+// Attach or detach an independent disk
+// Use the disk/action/attach or disk/action/detach links in a Vm to attach or detach an independent disk.
+// Reference: vCloud API Programming Guide for Service Providers vCloud API 30.0 PDF Page 164 - 165,
+// https://vdc-download.vmware.com/vmwb-repository/dcr-public/1b6cf07d-adb3-4dba-8c47-9c1c92b04857/
+// 241956dd-e128-4fcc-8131-bf66e1edd895/vcloud_sp_api_guide_30_0.pdf
+func (vm *VM) attachOrDetachDisk(diskParams *types.DiskAttachOrDetachParams, rel string) (Task, error) {
```

Comment:
Function doesn't check if diskParams argument is valid.

Reply:
I think it should be verified the arguments by vCloud API, because task of attachOrDetachDisk will be failed if those arguments are invalid.

Reply:
Yes it's true and yet I would say it is better to fail faster when you know exactly mandatory fields to be expected and provide more meaningful messages than xml parsing and other errors returned by VCD.

Reply:
I just added fields checking in the func.